core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -480,6 +480,34 @@ public static boolean validFileNameExcludingAstrix(String fileName) {
     }
 
     /**
+     * This does a very basic pass at validating that a header's value contains only expected characters according to RFC-5987, and those
+     * that it references.
+     * <p>
+     * https://tools.ietf.org/html/rfc5987
+     * <p>
+     * This is only expected to be used for assertions. The idea is that only readable US-ASCII characters are expected; the rest must be
+     * encoded with percent encoding, which makes checking for a valid character range very simple.
+     *
+     * @return {@code true} if the {@code value} is not obviously wrong.
+     */
+    public static boolean validHeaderValue(String value) {
```

Comment:
I don't think this method belongs on `Strings`, especially since it's only used in only place. I'd be fine with it as a static package-private method on `DeprecationRestChannel` so it's visible for testing.